src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -28,8 +28,12 @@ private ProjectId(string debugName)
             _debugName = debugName;
         }
 
-        internal ProjectId(Guid guid, string debugName)
+        public ProjectId(Guid guid, string debugName = null)
```

Comment:
I disagree on both points.  I prefer CreateA and CreateB because they give an indication on when each should be used (I'm the same way on overloads generally, I prefer to have different names indicating when to use each if they are really for different purposes).

In this specific case, the reason we never wanted the version with the GUID/DebugName exposed are:

1. To avoid dealing with GUID collisions.
2. To stop consumers from thinking that either of those were the identify, as opposed to object identity.

I think the new public API here *should* have a name which discourages actually using it, because we don't want people to use it.